CDTDatastore/query/CDTQProjectedDocumentRevision.h

```diff
@@ -28,12 +30,14 @@
 /**
  Initialise with a datastore so mutableCopy can return a full document.
  */
-- (id)initWithDocId:(NSString *)docId
-         revisionId:(NSString *)revId
-               body:(NSDictionary *)body
-            deleted:(BOOL)deleted
-        attachments:(NSDictionary *)attachments
-           sequence:(SequenceNumber)sequence
-          datastore:(CDTDatastore *)datastore;
+- (nullable instancetype)initWithDocId:(NSString *)docId
```

Comment:
Personally, I'm happy that the docs state that `NSObject`'s `init` can't return `nil`, and I can't see why they'd ever change it from just returning `self` which can surely never fail. Because of my lack of Swift knowledge I'm not sure what the impact is of having unnecessary `nullable` annotations other than them potentially propagating all round the code suggesting nullability that can't happen and presumably resulting in unnecessary checks in the code.